MoreLinq.Test/TriplewiseTest.cs

```diff
@@ -0,0 +1,49 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2008-2011 Jonathan Skeet. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class TriplewiseTest
+    {
+        [Test]
+        public void TriplewiseIsLazy()
+        {
+            new BreakingSequence<object>().Triplewise(BreakingFunc.Of<object, object, object, int>());
+        }
+
+        [TestCase(0)]
+        [TestCase(1)]
+        [TestCase(2)]
+        public void TriplewiseWithSequenceShorterThanTwo(int count)
+        {
+            var source = Enumerable.Range(0, count);
```

Comment:
You should use `AsTestingSequence()`
https://github.com/morelinq/MoreLINQ/blob/183ee85f914882a4d3e34f13dee60100c3d61a5d/MoreLinq.Test/TestingSequence.cs#L30